src/Returns/Domain/Payment/ReturnPayment.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelCom\Wms\Returns\Domain\Payment;
+
+readonly class ReturnPayment
+{
+    public function __construct(
+        public string $externalPaymentId,
+        public int $amount,
+        public Currency $currency,
+    ) {
+    }
+
+    public static function fromSnakeCaseArray(array $requestArray): self
```

Comment:
I'd avoid adding snake case in the name as it is always going to be in snake case. `from` or `create` I think give enough context.